e2e-test/src/test/java/io/cdap/plugin/normalize/stepsdesign/NormalizeTransform.java

```diff
@@ -0,0 +1,128 @@
+ * Normalize Plugin related step design.
+ */
+public class NormalizeTransform implements CdfHelper {
+  private static final Logger logger = LoggerFactory.getLogger(CdfPluginPropertiesActions.class);
+
+  @Then("Delete the output schema added")
+  public void deleteTheOutputSchemaAdded() {
+    NormalizeActions.clickDeleteField(0);
+  }
+
+  @Then("Verify Normalize plugin in-line error message for incorrect field mapping: {string}")
+  public void verifyNormalizePluginInLineErrorMessageForIncorrectFieldMapping(String fieldMapping) {
+    NormalizeTransform.verifyPluginPropertyInvalidInlineErrorMessage
+      ("fieldMapping",
+       PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_NORMALIZE_INVALID_FIELDMAPPING)
+         .replace("FIELD_MAPPING", PluginPropertyUtils.pluginProp(fieldMapping)));
+    NormalizeTransform.verifyPluginPropertyInvalidInlineErrorMessageColor("fieldMapping");
+  }
+
+  @Then("Verify Normalize plugin in-line error message for incorrect field to normalize: {string}")
+  public void verifyNormalizePluginInLineErrorMessageForIncorrectFieldToNormalize(String fieldNormalizing) {
+    NormalizeTransform.verifyPluginPropertyInvalidInlineErrorMessage
+      ("fieldNormalizing",
+       PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_NORMALIZE_INVALID_FIELDNORMALIZE)
+         .replace("FIELD_NORMALIZE", PluginPropertyUtils.pluginProp(fieldNormalizing)));
+    NormalizeTransform.verifyPluginPropertyInvalidInlineErrorMessageColor("fieldNormalizing");
+  }
+
+  public static void verifyPluginPropertyInvalidInlineErrorMessage(String propertyName,
+                                                                   String expectedInlineErrorMessage) {
+    WaitHelper.waitForElementToBeDisplayed(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName));
+    AssertionHelper.verifyElementDisplayed(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName));
+    AssertionHelper.verifyElementContainsText(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName),
+                                              expectedInlineErrorMessage);
+  }
+
+  public static void verifyPluginPropertyInvalidInlineErrorMessageColor(String propertyName) {
+    logger.info("Verify if plugin property: " + propertyName + "'s inline error message is shown in the expected color:"
+                  + " " + "#a40403");
+    String actualColor = ElementHelper.getElementColorCssProperty(NormalizeLocators.
+      locatePluginPropertyInvalidInlineError(propertyName));
+    String expectedColor = "#a40403";
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Select Normalize plugin property: output schema with value: {string}")
+  public void selectNormalizePluginPropertyOutputSchemaWithValue(String action) throws InterruptedException {
+      NormalizeActions.selectClearOutputSchema(PluginPropertyUtils.pluginProp(action));
+    }
+
+  @Then("Enter Normalize plugin FieldsMapping {string}")
```

Comment:
Enter Normalize plugin Fields to be Mapped {string}

Reply:
Change Done